### PR TITLE
Include sync status in Hamburg menu

### LINKED ISSUE #3357: Add sync status indicator for offline users (mobile and web)
Add an indicator to the hamburger menu so that users know when they last synced data and whether or not they have any data that needs to be synced. No specific count of documents is required at this time. More information in design spec here: https://docs.google.com/document/d/1fHrXcZZexM4DiXlEvABsYnM8upHN0IKBxMNkkXj8n8E/edit

Zeplin Files:
https://app.zeplin.io/project.html#pid=57d311edb055084578d006ec&sid=58e423770a9098e50f30e1b2
https://app.zeplin.io/project.html#pid=57d311edb055084578d006ec&sid=58e423780e7731f75cfe8ec8
https://app.zeplin.io/project.html#pid=57d311edb055084578d006ec&sid=58e423778617e25de61d283e

Original design issue: https://github.com/medic/medic-design/issues/48

### PR DESCRIPTION
**Pull request by alxndrsn**
Closes #3357

### CODE REVIEW

**Comment by alxndrsn** [comment]
Cancel that - I thought of some new stuff to add.

File: static/js/controllers/inbox.js [hunk 1]
```diff
@@ -53,20 +53,39 @@ var feedback = require('../modules/feedback'),
...
+          var delay = last ? (now - last) / 1000 : 'unknown';
+          $log.info('Replicate ' + update.direction + ' server successful with ' + delay + ' seconds since the previous successful replication.');
+        }
+
+        if (update.direction === 'to') {
+          if (update.state === 'not_required') {
+            $scope.replicationStatus.lastCompleted = now;
+          }
+          $scope.replicationStatus.current = update.state;
+          $scope.replicationStatus.icon = 'fa-' + SYNC_ICON[update.status];
```
Comment (garethbowen) [inline static/js/controllers/inbox.js]: Is it just me or is this adding too many "fa-"? Remove the prefix here.

File: templates/partials/header.html [hunk 2]
```diff
@@ -89,6 +89,20 @@
               <span translate>Log Out</span>
             </a>
           </li>
+
+          <li role="separator" class="divider" ng-if="!replicationStatus.disabled"></li>
+          <li role="presentation disabled" ng-show="!replicationStatus.disabled && replicationStatus.current !== 'unknown'">
+            <a>
+              Last sync <i class="fa fa-fw fa-refresh"></i><br>
```
Comment (garethbowen) [inline templates/partials/header.html]: Internationalisation!

File: templates/partials/header.html [hunk 3]
```diff
@@ -89,6 +89,20 @@
...
+              <span ng-bind-html="replicationStatus.lastCompleted | relativeDate"></span>
+            </a>
+          </li>
+          <li role="presentation disabled" ng-if="!replicationStatus.disabled" class="sync-status">
+            <a ng-class="replicationStatus.current">
+              <i class="fa fa-fw" ng-class="replicationStatus.icon"></i>
+              <span>{{'sync.status.' + replicationStatus.current | translate}}</span>
```
Comment (garethbowen) [inline templates/partials/header.html]: Don't concatenate keys - it makes it really hard to find if they're in use. Instead make `replicationStatus.current` have the full key, or have a `replicationStatus.key`.

Comment (garethbowen) [inline templates/partials/header.html]: Use `<span translate>{{replicationStatus.current}}</span>` - it reduces our watch count.

File: translations/messages-en.properties [hunk 4]
```diff
@@ -888,4 +888,8 @@ contact.profile.delivery_code.miscarriage = Miscarriage
 contact.profile.delivery_code.f = Facility birth
 contact.profile.delivery_code.s = Home birth with skilled attendant
 contact.profile.delivery_code.ns = Home birth without skilled attendant
-contact.profile.delivery_code.unknown = Delivery
\ No newline at end of file
+contact.profile.delivery_code.unknown = Delivery
+sync.status.in_progress=Currently syncing…
```
Comment (garethbowen) [inline translations/messages-en.properties]: I think you need spaces around the `=` operator, but I could be wrong...

File: static/js/controllers/inbox.js [hunk 5]
```diff
@@ -53,20 +53,39 @@ var feedback = require('../modules/feedback'),
...
           $log.debug('You have administrative privileges; not replicating');
           return;
         }
-        var now = Date.now();
-        var last = $scope.replicationStatus.lastSuccess[status.direction];
-        $scope.replicationStatus.lastSuccess[status.direction] = now;
-        var delay = last ? (now - last) / 1000 : 'unknown';
-        $log.info('Replicate ' + status.direction + ' server successful with ' + delay + ' seconds since the previous successful replication.');
+
+        var now;
```
Comment (garethbowen) [inline static/js/controllers/inbox.js]: Surely now must always be `Date.now()`? I think it's possible it'll be undefined at line 84. Just initialise it here, or inline it everywhere and remove the `now` var altogether.

File: static/css/inbox.less [hunk 6]
```diff
@@ -245,6 +245,15 @@ body {
   }
 }
 
+.sync-status {
+  .in_progress, .not_required {
+    color: green;
```
Comment (garethbowen) [inline static/css/inbox.less]: Always try and use existing variables for colours because then they'll be consistent with our palette rather than every commit adding a new colour. Also if we decide to change what colour should represent "ok" then we only need to change it in one place. The current variables are defined in variables.less. Consider using `@ok-color`.

File: static/css/inbox.less [hunk 7]
```diff
@@ -245,6 +245,15 @@ body {
...
+  }
+  .required {
+    color: red;
```
Comment (garethbowen) [inline static/css/inbox.less]: Same as above - try `@error-color`

File: translations/messages-en.properties [hunk 8]
```diff
@@ -888,4 +888,8 @@ contact.profile.delivery_code.miscarriage = Miscarriage
...
+sync.status.not_required=All reports synced
```
Comment (garethbowen) [inline translations/messages-en.properties]: Should this be "reports" or "docs"? I think docs is more technically correct but makes less sense to users maybe?

File: static/js/controllers/inbox.js (hunk 5 above)
Comment (alxndrsn) [inline static/js/controllers/inbox.js]: Not possible it will be undefined at line 84, but I agree it's hard to follow the code.  Seems logical to me to keep the value of `now` the same if assigning it in two different places - someone may think it's logical at a later date to compare the values at `lastSuccess` and `lastCompleted`, and my be surprised if they differ.

I'll move the assignment in line with the variable declaration.

File: static/js/controllers/inbox.js (hunk 1 above)
Comment (alxndrsn) [inline static/js/controllers/inbox.js]: # Fa-fa-fa-fa-fa-fa-fa-fa-fa-far better

File: templates/partials/header.html (hunk 2 above)
Comment (alxndrsn) [inline templates/partials/header.html]: Bien sûr!

File: templates/partials/header.html (hunk 3 above)
Comment (alxndrsn) [inline templates/partials/header.html]: 👍

Comment (alxndrsn) [inline templates/partials/header.html]: 👍 Good to know _someone's_ watching the watches.

File: translations/messages-en.properties (hunk 4 above)
Comment (alxndrsn) [inline translations/messages-en.properties]: Not sure they're required functionally, but definitely are for consistency.  Added.

File: static/css/inbox.less (hunk 6 above)
Comment (alxndrsn) [inline static/css/inbox.less]: `@ok-boss`

File: translations/messages-en.properties (hunk 8 above)
Comment (alxndrsn) [inline translations/messages-en.properties]: Good question.  I'm not sure if there was specific thinking behind this wording from the design team or not.

I agree that _docs_ is technically more correct, but perhaps _reports_ is more relevant to a user - he'll may be worrying whether the things he has submitted via the _Reports_ tab have been sent to the server.  Although... the _Reports_ tab is normally hidden, and he'll generally filling out reports via _Contacts_ or _Tasks_...

N.B. currently the only use of the word _docs_ in user-facing text is in the _About_ tab's db debug info, and soon to be in the boot screen...
